src/app/cli/commands.ts

```diff
@@ -0,0 +1,729 @@
+                        logger.error(
+                            `Failed to get session history: ${error instanceof Error ? error.message : String(error)}`
+                        );
+                    }
+                }
+                return true;
+            },
+        },
+        {
+            name: 'delete',
+            description: 'Delete a session',
+            usage: '/session delete <id>',
+            handler: async (args: string[], agent: SaikiAgent) => {
+                if (args.length === 0) {
+                    console.log(chalk.red('❌ Session ID required. Usage: /session delete <id>'));
+                    return true;
+                }
+
+                try {
+                    const sessionId = args[0]!; // Safe to assert non-null since we checked args.length
+
+                    const current = await getCurrentSessionInfo(agent);
+
+                    // Check if trying to delete current session
+                    if (sessionId === current.id) {
+                        console.log(
+                            chalk.yellow('⚠️  Cannot delete the currently active session.')
+                        );
+                        console.log(
+                            chalk.dim('   Switch to another session first, then delete this one.')
+                        );
+                        return true;
+                    }
+
+                    await agent.deleteSession(sessionId);
+                    console.log(chalk.green(`✅ Deleted session: ${chalk.bold(sessionId)}`));
+                } catch (error) {
+                    logger.error(
+                        `Failed to delete session: ${error instanceof Error ? error.message : String(error)}`
+                    );
+                }
+                return true;
+            },
+        },
+    ],
+    handler: async (args: string[], agent: SaikiAgent) => {
+        // Default to list if no subcommand
+        if (args.length === 0) {
+            const listSubcommand = sessionCommands.subcommands?.find((s) => s.name === 'list');
+            if (listSubcommand) {
+                return listSubcommand.handler([], agent);
+            }
+            return true;
+        }
+
+        const subcommand = args[0];
+        const subArgs = args.slice(1);
+
+        // Find matching subcommand
+        const subcmd = sessionCommands.subcommands?.find((s) => s.name === subcommand);
+        if (subcmd) {
+            return subcmd.handler(subArgs, agent);
+        }
+
+        console.log(chalk.red(`❌ Unknown session subcommand: ${subcommand}`));
+        console.log(
+            chalk.dim('Available subcommands: list, new, switch, current, history, delete')
+        );
+        return true;
+    },
+};
+
+/**
+ * Model management commands
+ */
+const modelCommands: CommandDefinition = {
+    name: 'model',
+    description: 'Manage AI models',
+    usage: '/model <subcommand> [args]',
+    aliases: ['m'],
+    subcommands: [
```

Comment:
let's also add help sub-command for /model that describes each of the allowed subcommands and when to use what